src/GraphQL/Utilities/Visitors/SchemaValidationVisitor.cs

```diff
@@ -129,6 +132,9 @@
                 throw new InvalidOperationException($"The field '{field.Name}' of an Interface type '{type.Name}' must be an output type.");
 
             ValidateFieldArgumentsUniqueness(field, type);
+
+            if (field.StreamResolver != null && type != schema.Subscription)
```

Comment:
Let's add the same check for the Resolver property of interface types.  We already determined that we must not have resolvers on interfaces because according to spec the interface must resolve to a type, and that way __typename can be resolved correctly, etc.